plugin/lighthouse/handler.go

```diff
@@ -61,26 +65,83 @@ func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 		return lh.nextOrFailure(state.Name(), ctx, w, r, dns.RcodeNameError, "Only services supported")
 	}
 
+	return lh.getDNSRecord(zone, state, ctx, w, r, pReq)
+}
+
+func (lh *Lighthouse) createARecords(ips []string, state request.Request) []dns.RR {
+	records := make([]dns.RR, 0)
+
+	for _, ip := range ips {
+		record := &dns.A{Hdr: dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass(), Ttl: lh.ttl}, A: net.ParseIP(ip).To4()}
+		log.Debugf("rr is %v", record)
+		records = append(records, record)
+	}
+
+	return records
+}
+
+func (lh *Lighthouse) createSRVRecords(record *serviceimport.DNSRecord, state request.Request, pReq recordRequest, zone string) []dns.RR {
+	reqPorts := make([]mcsv1a1.ServicePort, 0)
+	records := make([]dns.RR, 0)
```

Comment:
Move this right before L110:

```suggestion
	records := make([]dns.RR, len(reqPorts))
```

Reply:
Done.